app/main.py

```diff
@@ -0,0 +1,48 @@
+# from pydantic import BaseModel
+from fastapi import Depends, FastAPI, Request
+from fastapi.responses import HTMLResponse
+from fastapi.staticfiles import StaticFiles
+from fastapi.templating import Jinja2Templates
+from sqlalchemy import select
+from sqlalchemy.orm import Session
+
+from .models import Base, Product, SessionLocal, engine
+
+app = FastAPI()
+templates = Jinja2Templates(directory="templates")
+app.mount("/static", StaticFiles(directory="static"), name="static")
+
+@app.on_event("startup")
+def startup():
+    Base.metadata.create_all(bind=engine)
+
+def get_db():
+    db = SessionLocal()
```

Comment:
For a moment I thought, there is `SessionLocal` in SQLAlchemy.

What you'd probably want to do is to use a context manager directly. That way, you will definitely avoid Connection Pool's issues (sessions left open, etc.).

```python
with SessionLocal() as async_session:
    # async_session.do_stuff...
    await session.commit()
```

That way, you won't even need `get_db()`. Well, unless you want to wrap it with something more.


P.S. It'd be incorrect (too general) to call the variable `db`, since it's a DB-session, not a DB itself.